src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8243,6 +8243,12 @@ internal static bool IsReparsePointLikeSymlink(FileSystemInfo fileInfo)
 
             WIN32_FIND_DATA data = default;
             string fullPath = Path.TrimEndingDirectorySeparator(fileInfo.FullName);
+
+            if (fullPath.Length > MAX_PATH)
+            {
+                fullPath = "\\\\?\\" + fullPath;
```

Comment:
There are 2 things we need to check here before we can add the prefix.

* The prefix isn't already present - I'm not sure if `FileSystemInfo.FullName` preserves the `\\?\` if it was first created with that prefix
* When dealing with a UNC path we need to do `\\?\UNC\{server}\{share}`, essentially replace the first `\\` with `\\?\UNC`

I'm also not sure if pwsh/.NET is meant to support the `\\.\` DosDevice prefix as well as `\\?\`, if so then it also needs to check for that prefix.

Finally I think that the prefix should always be added rather than when the path exceeds `MAX_PATH`. This ensures the code always runs the same no matter what path is being used and any future bugs that might occur occur in all situations (thus gets picked up more easily/in betas) rather than just in the edge cases.